rpc/server/rpc_faucet.go

```diff
@@ -92,24 +101,48 @@ func (f *faucet) Claim(
 		}
 	}()
 
-	addrPubHash, err := types.NewAddressFromPubKey(f.account.PrivateKey().PubKey())
-	if err != nil {
-		return newClaimResp(-1, err.Error()), nil
+	pr, ok := peer.FromContext(ctx)
+	logger.Warnf("peer form context pr: %+v, ok : %t", pr, ok)
+	inWhiteList := false
+	for _, v := range f.faucetConfig.IPList {
+		if strings.HasPrefix(pr.Addr.String(), v) {
```

Comment:
not strictly.
to be true if ip is 192.168.0.123:2335 and pr.Addr is 192.168.0.12